### ISSUE TITLE
Resolve USWDS/Template Strategy

### ISSUE THREAD

**Issue by wslack** [18F/federalist#1262]
Update 1/8/17:

 - Level of Effort: Medium
 - Implementation Sketch (if higher than a "low" effort): mostly completed already
 - Definition of Done: We have an agreed-upon template strategy with prioritized tasks, including making new templates.

Per https://github.com/18F/guides-template/issues/84#issuecomment-340086629

- [x] Will / Scott meet sync or async on templates.
- [x] Will / Scott align on plan for the USWDS template.
- [x] Will / Scott resolve if there is a guides template need (unblocks another issue)

Goals: Align on strategy, consolidate functionality, make templates easier to update.

This will be done when: there is a clear strategy for Federalist template work.

Federalist Template Goals working doc - https://docs.google.com/document/d/1HfJmjbQ7uFh7z0tB6T0d2RiIg5vLYTvTl0bnZFzHPvs/edit#heading=h.xr1nr21pdcww

**Comment by line47** [comment]
Here are my thoughts around the template strategy ATM:

I took a look at the 4 templates that are currently available in federalist and started [this Mural](https://app.mural.ly/t/gsa6/m/gsa6/1510166920317/1d31337586b8fe60c3a1e1fd18182dd4823cd165) to make note of unique functionality or design elements from the various templates. 

- [U.S. Web Design Standards Template](https://federalist-uswds-template.18f.gov/)
- [Multi Page Template](https://federalist-modern-team-template.18f.gov/)
- [Single Page Template](https://federalist-landing-template.18f.gov/)
- [Basic Report Template](https://federalist-proxy.app.cloud.gov/site/18f/federalist-report-template/)
---
Thinking about components/functionality for the following:

- [ ] **USA banner**  
(can be turned on or off in the config.yml) 
- [ ] **Header**
(can set type of header in the header.yml - slim, basic or extended)
- [ ] **Google analytics**
(set in the _config.yml) - takes a look at how the guides style is doing it https://github.com/18F/guides-style/blob/master/_includes/analytics.html How to add to the config.yml - https://github.com/18F/guides-style#additional-features
- [ ] **Search**
(opt in) Offer Jekyll pages API search and saerch.gov search  Take documentation from https://github.com/18F/guides-style#search

**Comment by wslack** [comment]
And as promised, here's v0.1 of the template strategic goals for your feedback!

## Template Goals

#### Priority 1
- Demonstrate Federalist's value at a glance for business development
- Demonstrate types of Federalist sites via code and live presentation

#### Priority 2
- Make it easier to launch common types of sites that Federalist customers want
  - includes good documentation in the templates on how to use/modify them
- Demonstrate how to organize the Jekyll/Hugo code of Federalist sites
- Provide configuration options that people can opt into or out of

#### Priority 3
 - Demonstrate capabilities of the USWDS
 - Provide modular components
 - show different types of headers/footers that can be easily changed out for each other
   - low priority because of the level of effort to keep everything synced between different types of templates

**Comment by line47** [comment]
[Federalist report template](https://federalist-proxy.app.cloud.gov/preview/18f/uswds-jekyll/header-slim/) built utilizing the USWDS-Jekyll theme

**Comment by davidewarren** [comment]
New Goals - 11/27 Sprint Planning

- [ ] Clean up the Federalist Template Goals working doc EOD Wednesday - https://docs.google.com/document/d/1HfJmjbQ7uFh7z0tB6T0d2RiIg5vLYTvTl0bnZFzHPvs/edit#heading=h.xr1nr21pdcww

- [ ] Front end guild Federalist Template discussion

- [ ] Final decision for Federalist Template direction

**Comment by line47** [comment]
Takeaways from 12/1/2017 Front end rally 🏁 

- Repos should have one canonical (Master) branch.
- It would be good to have a curated "site set up flow" from a primary USWDS template.
- Explore config options with the USWDS template
- Document how to use (offer snippets for quick config altering)
- Categories of templates - Call out maintained/unmaintained templates
- 18F doesn't have a specific best practice w/r/t Jekyll site management.
  - **Recommend:** Follow Jekyll guidance for best practices
- It's ok to have multiple templates when site differences can't all be contained within config. 
  - **Recommend:** The 4 Federalist templates can be adjusted in a config
- Maintaining multiple branches and codebases is :thumbsdown:.

**Comment by line47** [comment]
**Plan for moving forward** 
- Keep USWDS-Jekyll as the source for all USWDS components and functionality
  - Make a template from the USWDS-Jekyll repo that contains all the files (for experienced web dev team/individual)


- Use federalist-uswds-template repo to provide multiple configuration options. 
All of these options only have the files that need to be modified and do not contain everything that is part of USWDS-Jekyll. If you want all the files you should use the template that points at the USWDS-Jekyll repo
   - Add any custom functionality or components to the master branch on the federalist-uswds-template making the component available for other configurations
  - Report config template
  - Team site config template
  - One page config template
  - USWDS config template

**Comment by wslack** [comment]
I don't quite follow.

Is the hypothesis here that the report, team site, and one pager can all be accomplished only by changing the config file and nothing else?

**Comment by line47** [comment]
Yeah, that's exactly right 👍

**Comment by wslack** [comment]
Ok. One other question - what does this imply about turning plainlanguage.gov into a template and the other blocked template issues?

**Comment by wslack** [comment]
(I want to unblock all of those by EOD if possible)

**Comment by line47** [comment]
I'd say turning something like plainlanguage.gov would be a site categorized as a template is not maintained by the Federalist team. Thus, could be lacking documentation and guides for how to use/extend that particular template.

**Comment by wslack** [comment]
What would you do with https://github.com/18F/federalist/issues/1239?

**Comment by line47** [comment]
Make is as a template but caveat that it’s not a federally maintained template

**Comment by wslack** [comment]
Cool. As long as we're on the same page that the new build we're doing for Jay is a separate type of site/experiment, I think this can be moved to `Done?`.

**Comment by line47** [comment]
I agree! 💯

**Comment by wslack** [comment]
Ok, my dueout on Monday is to re-sort the issues based on this. Which template do you want to replace with the centralized model first?

**Comment by line47** [comment]
The report template is basically already created - https://github.com/18F/federalist-uswds-template/tree/report-config 

Preview - https://federalist-proxy.app.cloud.gov/preview/18f/federalist-uswds-template/report-config/

So let's start there

**Comment by wslack** [comment]
Cool. is there any issue open that for that already?

**Comment by line47** [comment]
I did not create an issue for that, not sure if there is a previous issue around that work or not. I'm happy to create one - just let me know 👍

**Comment by wslack** [comment]
I can check later on today but let's assume there's not one now, so you can track what's been delivered for the new structure so far.
